### DISCUSSION TITLE
Incorrect Report Directory Creation

### DISCUSSION THREAD

**Discussion by RomarioRa94** [ortoniKC/ortoni-report#32] [Q&A] [answered]
**Summary:**
When running Playwright tests with different commands, the report generation directory behaves inconsistently.

**Description:**
When running the tests using the command url=BASE_URL_DEV npx playwright test tests/api --project=Headed --workers=1, the report is being created directly in the project directory under `/Users/romariora/Documents/GitHub/xyz/report/index.html, instead of inside the test folder as expected`

In contrast, running the test with the Playwright test runner directly generates the report correctly at `/Users/romariora/Documents/GitHub/xyz/test/report/index.html`

```
// Determine the root directory based on the current working directory
const rootDir = process.cwd().match(/test$/)
  ? process.cwd()
  : path.join(process.cwd(), "test");

const report = {
  filename: path.join(rootDir, "ortoni-report.html"),
  logo: path.join(rootDir, "files/logo.png"),
  title: "Test Report",
  showProject: true,
  projectName: "Project",
  testType: "Functional",
  authorName: "Romario",
  preferredTheme: "light",
  base64Image: true,
};

reporter: [["ortoni-report", report], ["dot"]];
```

Passing the `folderPath` directly behaves the same way. Here, I've added a setup for a working directory, which works up to version 2.0.2. However, starting from version 2.0.3, this is also not functioning correctly.

**Comment by ortoniKC** [comment]
Starting from version 2.0.3 folder path and file name are handled using in built node path. 
Use as a string it should work fine I believe.

```
folderPath: "test", 
filename: "index",
```

**Comment by RomarioRa94** [comment]
@ortoniKC , thanks for the quick heads-up. I have already tried this method, and the behavior remains the same.

**Comment by ortoniKC** [comment]
Is it possible to give a reproducible code? 
As I'm not able to reproduce it.

**Comment by ortoniKC** [comment]
Playwright version: 2.0.4
Here is my observation: 
1. When using path - report is stored in 'Y:\code\my-playwright-report\playwright-report\Y:\code\my-playwright-report\tests\ortoni-report.html'
As the report uses abosulte path from the root it is trying to create another abs path.
2. When passing folderPath, filename & logo as string it works as expected.

Passing env variable should not effect the execution and generating report anyway.
Please delete the node_modules folder, add the 2.0.4 package and try. 
It must work, if not. Please provide a reproducible repo, so that i can figure it out.

Thanks.

![Screenshot 2024-10-18 124309](https://github.com/user-attachments/assets/06d77a8a-53af-4bab-a62c-a0f333babece)
![image](https://github.com/user-attachments/assets/b00b2202-f19c-4a0a-afd1-1d9c7e1527b5)

**Comment by ortoniKC** [comment]
Closing for now as no response was received.

**Comment by RomarioRa94** [comment]
The issue still persists, so I decided to stick with version 2.0.2.

**Reply by ortoniKC** [reply, marked as answer]
You're missing all the latest features, 
If you can send me a reproducible code then I can definitely fix it.

**Comment by RomarioRa94** [comment]
Thanks @ortoniKC  , The issue has been resolved.
